File: Data Structures & Algorithms/anagram-groups/submission-2.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        int [] hash = new int[26];
        List<List<String>> group = new ArrayList<List<String>>();
        
        for (int j = 0; j < strs.length; j++) {
            String word = strs[j]; 
            if (word == null) continue; 
            // Iterate through each, if an anagram is foudn add to list
            // At the end, add word itself 
            strs[j] = null;
            List<String> inner = new ArrayList<String>(); 
            inner.add(word); 
            
            for (int i = j + 1; i < strs.length; i++) {
                if (strs[i] != null && isAnagram(strs[i], word)) {
                    inner.add(strs[i]);
                    strs[i] = null;
                }
            }
            group.add(inner);

            
        }
        return group;
    }





    public static boolean isAnagram(String a, String b) {
        
        if (a.length() != b.length()) {
            return false; 
        }

        int [] hash = new int [26]; 

        for (int i = 0; i<a.length(); i++) {
            hash[a.charAt(i) - 'a']++;
            hash[b.charAt(i) - 'a']--;
        }


        for (int i : hash) {
            if (i != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashmap_s = {}
        hashmap_t = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if hashmap_s.get(s[i]):
                hashmap_s[s[i]] = hashmap_s.get(s[i]) + 1
            else: 
                hashmap_s[s[i]] = 1

            if hashmap_t.get(t[i]):
                hashmap_t[t[i]] = hashmap_t.get(t[i]) + 1
            else: 
                hashmap_t[t[i]] = 1

            
        if hashmap_s == hashmap_t:
            return True
        else:
            return False